setup_dev.py

```diff
@@ -0,0 +1,532 @@
+        f"-DBUILD_SHARED_LIBS={'ON' if args.llvm_shared else 'OFF'}",
+        f"-DLLVM_DIR={llvm_cmake_dir}",
+        "-DLLVM_INSTALL_TOOLCHAIN_ONLY=OFF",
+        "-DLLVM_BUILD_TOOLS=ON",
+        "-DMLIR_ENABLE_BINDINGS_PYTHON=ON",
+    ]
+    mlir_configure_cmake_args.extend(build_type_args)
+    log_verbose(f"+ Configuring MLIR: {' '.join(mlir_configure_cmake_args)}")
+    subprocess.check_call(mlir_configure_cmake_args)
+
+  # MLIR build step.
+  mlir_build_cmake_args = [
+      "cmake",
+      "--build",
+      str(mlir_build_dir),
+  ]
+  log_verbose(f"+ Building MLIR: {' '.join(mlir_build_cmake_args)}")
+  subprocess.check_call(mlir_build_cmake_args)
+
+  # Locate MLIR cmake directories.
+  mlir_cmake_dir = mlir_build_dir / "lib" / "cmake" / "mlir"
+  verify_cmake_dir(mlir_cmake_dir)
+
+  # Write llvm dep config.
+  config_lines = [
+      f'set(LLVM_DIR "{llvm_cmake_dir}" CACHE STRING "Auto Configured CMake dir" FORCE)',
+      f'set(LLD_DIR "{lld_cmake_dir}" CACHE STRING "Auto Configured CMake dir" FORCE)',
+      f'set(Clang_DIR "{clang_cmake_dir}" CACHE STRING "Auto Configured CMake dir" FORCE)',
+      f'set(MLIR_DIR "{mlir_cmake_dir}" CACHE STRING "Auto Configured CMake dir" FORCE)',
+      f'set(IREE_BUILD_BUNDLED_LLVM OFF CACHE BOOL "Auto Configured setting" FORCE)',
+      f'set(IREE_COMPILER_BUILD_SHARED_LIBS {"ON" if args.llvm_shared else "OFF"} CACHE BOOL "Auto Configured setting" FORCE)',
+  ]
+  config.write_config_cmake(".llvm_dep_config.cmake", "\n".join(config_lines))
+
+
+def build_internal_llvm(args, config: Config):
+  # Write llvm dep config.
+  config_lines = [
+      f'set(IREE_BUILD_BUNDLED_LLVM ON CACHE BOOL "Auto Configured setting" FORCE)',
+      f'set(IREE_COMPILER_BUILD_SHARED_LIBS {"ON" if args.llvm_shared else "OFF"} CACHE BOOL "Auto Configured setting" FORCE)',
+  ]
+  config.write_config_cmake(".llvm_dep_config.cmake", "\n".join(config_lines))
+
+
+def main(args):
+  config = Config(args)
+  log_verbose(f"- generating {config.toolchain_config_file_name}")
+  setup_toolchain(args, config)
+
+  # Build LLVM and MLIR.
+  if args.llvm_external_build:
+    build_external_llvm(args, config)
+  else:
+    build_internal_llvm(args, config)
+
+  next_steps = f"""# Initial CMake configure.
+cmake -GNinja -S{repo_root} -B{config.iree_build_dir} {' '.join(build_config_to_args(args.config, "iree"))} -DIREE_BUILD_PYTHON_BINDINGS=ON
+# Build project.
+cmake --build {config.iree_build_dir}
+# Build project and run tests.
+cmake --build {config.iree_build_dir} --target iree-run-tests
+"""
+
+  # Print a cheat-sheet.
+  cheatsheet = f"""# IREE Development Cheat Sheet
+
+## Initial build
+
+```
+{next_steps}
+```
+
+## Rebuild LLVM/LLD/Clang/MLIR
+
+```
+# Rebuild LLVM+LLD+Clang
+cmake --build {config.llvm_build_dir}
+# Rebuild MLIR
+cmake --build {config.mlir_build_dir}
+```
+"""
+  cheatsheet_path = repo_root / "dev_cheatsheet.md"
+  with open(repo_root / "dev_cheatsheet.md", "wt") as f:
```

Comment:
Does this (`repo_root / "subdir"`) work in Python correctly? Usually I'd do `os.path.join(...`)